.travis.yml

```diff
@@ -1,32 +1,44 @@
-
+dist: trusty
 language: cpp
 sudo: required
 
 addons:
   apt:
     sources:
       - ubuntu-toolchain-r-test
-      - llvm-toolchain-precise-3.7
+      - llvm-toolchain-precise
     packages:
       - g++-4.8
-      - clang-3.7
+      - clang-3.8
       - libpng-dev
       - libjpeg8-dev
       - libtiff4-dev
       - libtbb-dev
       - libatlas-base-dev
+      - libboost-atomic1.55-dev
+      - libboost-chrono1.55-dev
+      - libboost-filesystem1.55-dev
+      - libboost-date-time1.55-dev
+      - libboost-program-options1.55-dev
+      - libboost-serialization1.55-dev
+      - libboost-system1.55-dev
+      - libboost-thread1.55-dev
+      - libboost-timer1.55-dev
 
 matrix:
   include:
     - compiler: gcc
-      env: CXX_COMPILER=g++-4.8 C_COMPILER=gcc-4.8 TOOLSET=gcc VISUAL_DEBUG=ON CCTAG_SERIALIZE=ON WITH_CUDA=ON
+      env: CXX_COMPILER=g++-4.8 C_COMPILER=gcc-4.8 TOOLSET=gcc WITH_CUDA=ON CUDA_VERSION_MAJOR="7" CUDA_VERSION_MINOR="0" CUDA_PKG_LONGVERSION="${CUDA_VERSION_MAJOR}.${CUDA_VERSION_MINOR}-28" CUDA_PKG_VERSION="${CUDA_VERSION_MAJOR}-${CUDA_VERSION_MINOR}"
+    - compiler: gcc
+      env: CXX_COMPILER=g++-4.8 C_COMPILER=gcc-4.8 TOOLSET=gcc WITH_CUDA=ON CUDA_VERSION_MAJOR="7" CUDA_VERSION_MINOR="5" CUDA_PKG_LONGVERSION="${CUDA_VERSION_MAJOR}.${CUDA_VERSION_MINOR}-18" CUDA_PKG_VERSION="${CUDA_VERSION_MAJOR}-${CUDA_VERSION_MINOR}"
     - compiler: gcc
-      env: CXX_COMPILER=g++-4.8 C_COMPILER=gcc-4.8 TOOLSET=gcc VISUAL_DEBUG=OFF CCTAG_SERIALIZE=ON WITH_CUDA=OFF
+      env: CXX_COMPILER=g++-4.8 C_COMPILER=gcc-4.8 TOOLSET=gcc WITH_CUDA=ON CUDA_VERSION_MAJOR="8" CUDA_VERSION_MINOR="0" CUDA_PKG_LONGVERSION="${CUDA_VERSION_MAJOR}.${CUDA_VERSION_MINOR}.61-1" CUDA_PKG_VERSION="${CUDA_VERSION_MAJOR}-${CUDA_VERSION_MINOR}"
```

Comment:
Is it not better to add a new variant?
In order to keep one build variant with "VISUAL_DEBUG=ON CCTAG_SERIALIZE=ON"?
To ensure that the code with debug exports is builded once and not broken by new changes.